src/CompositeRheologies.jl

```diff
@@ -0,0 +1,794 @@
+    str = print_rheology_matrix(g)
+    println.(str)
+    
+
+    return nothing
+end
+
+function show(io::IO, a::Parallel)
+    println(io,"Parallel:   ")  
+
+    # Compose a string with rheological elements, so we have an overview in the REPL
+    #str = create_rheology_string("",a)
+    str = print_rheology_matrix(a)
+    println.(str)
+
+    return nothing
+end
+
+# COMPUTE STRAIN RATE
+"""
+    compute_εII(v::Parallel{T,N}, τII, args; tol=1e-6, verbose=false, n=1)
+
+Computing `εII` as a function of `τII` for a Parallel elements is (usually) a nonlinear problem
+"""
+function compute_εII(
+    v::Parallel{T,N}, 
+    τII::_T, 
+    args; 
+    tol=1e-6, verbose=false, n=1
+) where {T,N,_T}
+    εII = local_iterations_τII(v, τII, args; tol=tol, verbose=verbose, n=n)
+    return εII
+end
+
+"""
+    compute_εII(v::CompositeRheology{T,N}, τII, args; tol=1e-6, verbose=false, n=1)
+
+Computing `εII` as a function of `τII` for a composite element is the sum of the individual contributions
+"""
+@generated  function compute_εII(
+    v::CompositeRheology{T,N}, 
+    τII::_T, 
+    args; 
+    tol=1e-6, verbose=false
+) where {T,_T,N}
+    quote
+        Base.@_inline_meta
+        εII = zero(_T)
+        Base.Cartesian.@nexprs $N i ->
+            εII += compute_εII(v.elements[i], τII, args)
+    end
+end
+
+
+# As we don't do iterations, this is the same
+function compute_εII_AD(v::CompositeRheology, τII, args; tol=1e-6, verbose=false)
+    return  compute_εII(v, τII, args)
+end
+
+# Here we do need to do iterations
+function compute_εII_AD(v::Parallel, τII, args; tol=1e-6, verbose=false)
+    return local_iterations_τII_AD(v, τII, args; tol=tol, verbose=verbose)
+end
+
+
+# COMPUTE DEVIATORIC STRESS
+function compute_τII(v::CompositeRheology{T,N,0}, εII, args; tol=1e-6, verbose=false) where {T,N}
+    # A composite rheology case with no parallel element; iterations for τII
+    τII = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose)
+    return τII
+end
+
+"""
+    τII = compute_τII(v::CompositeRheology{T,N}, εII, args; tol=1e-6, verbose=false)
+    
+"""
+function compute_τII(v::CompositeRheology{T,N}, εII, args; tol=1e-6, verbose=false, τ_initial=nothing, ε_init=nothing) where {T,N}
+    # A composite rheology case with parallel elements
+    τII = local_iterations_εII(v, εII, args; tol=tol, verbose=verbose, τ_initial=τ_initial, ε_init=ε_init)
+    return τII
+end
+
+# For a parallel element, τII for a given εII is the sum of each component
+@generated  function compute_τII(
+    v::Parallel{T,N}, 
+    εII::_T, 
+    args;
+    tol=1e-6, verbose=false
+) where {T,_T,N}
+    quote
+        Base.@_inline_meta
+        τII = zero(_T)
+        Base.Cartesian.@nexprs $N i ->
+            τII += compute_τII(v.elements[i], εII, args)
+    end
+end
+compute_τII_AD(v::Parallel{T,N}, εII::_T, args; tol=1e-6, verbose=false) where {T,N,_T} = compute_τII(v, εII, args) 
+
+
+@generated  function compute_τII_i(
```

Comment:
I believe we don't need this function anymore

Reply:
correct, removed